qemu/tests/qemu_guest_agent.py

```diff
@@ -1849,6 +1853,124 @@ def gagent_check_with_migrate(self, test, params, env):
         error_context.context("Verify if guest agent works.", logging.info)
         self.gagent_verify(self.params, self.vm)
 
+    @error_context.context_aware
+    def gagent_check_umount_frozen(self, test, params, env):
+        """
+        Umount file system while fs freeze.
+
+        Steps:
+        1) boot guest with a new data storage.
+        2) format disk and mount it.
+        3) freeze fs.
+        4) umount fs/offline the volume in guest
+        5) thaw fs.
+        6) mount fs and online it again.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+        def wrap_windows_cmd(cmd):
+            """
+            add header and footer for cmd in order to run it in diskpart tool.
+
+            :param cmd: cmd to be wrapped.
+            :return: wrapped cmd
+            """
+            disk = "disk_" + ''.join(random.sample(string.ascii_letters +
+                                                   string.digits, 4))
+            cmd_header = "echo list disk > " + disk
+            cmd_header += " && echo select disk %s >> " + disk
+            cmd_footer = " echo exit >> " + disk
+            cmd_footer += " && diskpart /s " + disk
+            cmd_footer += " && del /f " + disk
+            cmd_list = []
+            for i in cmd.split(";"):
+                i += " >> " + disk
+                cmd_list.append(i)
+            cmd = " && ".join(cmd_list)
+            return " && ".join([cmd_header, cmd, cmd_footer])
+
+        session = self._get_session(params, self.vm)
+        image_size_stg0 = params["image_size_stg0"]
+        is_rhel6 = params.get("is_rhel6")
+
+        error_context.context("Format the new data disk and mount it.",
+                              logging.info)
+        if params.get("os_type") == "linux":
+            disk_data = list(utils_disk.get_linux_disks(session).keys())
+            mnt_point = utils_disk.configure_empty_disk(
+                session, disk_data[0], image_size_stg0, "linux",
+                labeltype="msdos")
+            partition = "/dev/%s" % mnt_point[0].split("/")[-1]
```

Comment:
I don't understand this usage, is `mnt_point[0]` like `/dev/sda` or directly `sda` something? Which case you need `split("/")[-1]` and add `/dev` before it?

Reply:
Modified.